src/PSAppDeployToolkit/Public/Set-ADTDeferHistory.ps1

```diff
@@ -54,12 +61,16 @@ function Set-ADTDeferHistory
 
         [Parameter(Mandatory = $false)]
         [AllowEmptyString()]
-        [System.String]$DeferDeadline
+        [System.String]$DeferDeadline,
+
+        [Parameter(Mandatory = $false)]
+        [ValidateNotNullOrEmpty()]
+        [System.UInt32]$DeferRunInterval
     )
 
     try
     {
-        (Get-ADTSession).SetDeferHistory($(if ($PSBoundParameters.ContainsKey('DeferTimesRemaining')) { $DeferTimesRemaining }), $DeferDeadline)
+        (Get-ADTSession).SetDeferHistory($(if ($PSBoundParameters.ContainsKey('DeferTimesRemaining')) { $DeferTimesRemaining }), $DeferDeadline, $(if ($PSBoundParameters.ContainsKey('DeferRunInterval')) { $DeferRunInterval }), $(if ($PSBoundParameters.ContainsKey('DeferRunInterval')) { (Get-ADTUniversalDate) }))
```

Comment:
While it's not overly expensive, we're doing a dictionary check twice with `$PSBoundParameters.ContainsKey('DeferRunInterval')`. Can we either cache the result, or do something like:
```PowerShell
if ($PSBoundParameters.ContainsKey('DeferRunInterval'))
{
    (Get-ADTSession).SetDeferHistory($(if ($PSBoundParameters.ContainsKey('DeferTimesRemaining')) { $DeferTimesRemaining }), $DeferDeadline, $DeferRunInterval, (Get-ADTUniversalDate))
}
else
{
    (Get-ADTSession).SetDeferHistory($(if ($PSBoundParameters.ContainsKey('DeferTimesRemaining')) { $DeferTimesRemaining }), $DeferDeadline, $null, $null)
}